src/actions/posts.ts

```diff
@@ -1196,31 +1234,33 @@ function lastPostActions(post: Post, websocketMessageProps: any) {
             },
         ];
 
-        dispatch(batchActions(actions));
+        await dispatch(batchActions(actions));
 
         if (shouldIgnorePost(post)) {
             return;
         }
 
         let markAsRead = false;
         let markAsReadOnServer = false;
-        if (
-            post.user_id === getCurrentUserId(state) &&
-            !isSystemMessage(post) &&
-            !isFromWebhook(post)
-        ) {
-            markAsRead = true;
-            markAsReadOnServer = false;
-        } else if (post.channel_id === getCurrentChannelId(state)) {
-            markAsRead = true;
-            markAsReadOnServer = true;
+        if (!isManuallyUnread(getState(), post.channel_id)) {
+            if (
+                post.user_id === getCurrentUserId(state) &&
+                !isSystemMessage(post) &&
+                !isFromWebhook(post)
+            ) {
+                markAsRead = true;
+                markAsReadOnServer = false;
+            } else if (post.channel_id === getCurrentChannelId(state)) {
+                markAsRead = true;
+                markAsReadOnServer = true;
+            }
         }
 
         if (markAsRead) {
-            dispatch(markChannelAsRead(post.channel_id, undefined, markAsReadOnServer));
-            dispatch(markChannelAsViewed(post.channel_id));
+            await dispatch(markChannelAsRead(post.channel_id, undefined, markAsReadOnServer));
```

Comment:
I was running into issues testing this code, and I thought at the time that it was caused by those running asynchronously before the code that was checking the results. I don't think that ended up being the cause of my testing problems, but I left it in just in case.